src/hooks/useSwapOptions.ts

```diff
@@ -138,7 +133,13 @@ function useSwapOptions(
     return [undefined, false, undefined, fetchOptions];
   }
 
-  return [result || prevResult, isLoading, error, fetchOptions];
+  let resultToReturn = result || prevResult;
+
+  if (sorting === 'least-gas' && resultToReturn) {
+    resultToReturn = [...resultToReturn].sort((a, b) => (a.gas.estimatedCost.lt(b.gas.estimatedCost) ? -1 : 1));
```

Comment:
We should probably React.useMemo this, so it does not recalculate it on every re-render

Reply:
I don't think that is an issue, the list of routes is small and the check is very simple. Imo the load of useMemo would be worse than the check itself.